HOTIFX: streams system test do not start up correctly

mjsax opened the pull request.

mjsax committed:
HOTIFX: streams system test do not start up correctly

mjsax commented:
All 4 system tests did pass locally. Will trigger a Jenkins run, too.

@enothereska @dguy @guozhangwang

mjsax commented:
(not started yet though)
https://jenkins.confluent.io/job/system-test-kafka-branch-builder/683/
https://jenkins.confluent.io/job/system-test-kafka-branch-builder/684/

mjsax commented:
I guess this should go to `0.10.2`, too.

guozhangwang commented on streams/src/test/java/org/apache/kafka/streams/systemtest/ShutdownDeadlockTest.java:
```diff
@@ -14,7 +14,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.apache.kafka.streams.smoketest;
```
Any particular reason you wanted to change the package name?

mjsax replied:
As discussed here: https://github.com/apache/kafka/pull/2403/files#r97225524

guozhangwang replied:
I see. In that case could we just move to `org.apache.kafka.streams.tests`? The rationale is that we do not have a `systemtest` sub-package in other kafka modules yet and all `kafkatest/services` (i.e. the ducktape system tests repo) either trigger some script to start the service in-directly or depend on its mock in `o.a.k.[package].tests`.

mjsax replied:
Ack

dguy commented on streams/src/test/java/org/apache/kafka/streams/perf/SimpleBenchmark.java:
```diff
@@ -98,7 +99,7 @@ public SimpleBenchmark(File stateDir, String kafka) {
     public static void main(String[] args) throws Exception {
         String kafka = args.length > 0 ? args[0] : "localhost:9092";
-        String stateDirStr = args.length > 1 ? args[1] : "/tmp/kafka-streams-simple-benchmark";
+        String stateDirStr = args.length > 1 ? args[1] : TestUtils.tempDirectory("kafka-stream-system-test/kafka-streams-simple-benchmark").getAbsolutePath();
```
Any reason why you didn't just use `TestUtils.tempDirectory()` ? i.e., does it need to have `kafka-streams-system-test/kafka-streams-simple-benchmark` ?

dguy commented on streams/src/test/java/org/apache/kafka/streams/systemtest/SmokeTestDriver.java:
```diff
@@ -78,7 +78,7 @@ int next() {
         final String kafka = "localhost:9092";
         final String zookeeper = "localhost:2181";
-        final File stateDir = TestUtils.tempDirectory();
+        final File stateDir = TestUtils.tempDirectory("kafka-stream-system-test");
```
Why do we need this?

enothereska commented:
Awesome, thanks @mjsax LGTM

mjsax committed:
GitHub comments

mjsax commented:
@dguy @enothereska @guozhangwang

Addressed your comments. All 4 tests did pass locally. Triggered one more run on Jenkins:

(not started yet though)
https://jenkins.confluent.io/job/system-test-kafka-branch-builder/686/
https://jenkins.confluent.io/job/system-test-kafka-branch-builder/687/

mjsax commented:
retest this please.

guozhangwang commented:
683/4/6/7 all passed.

guozhangwang commented:
Merged to trunk, and cherry-picked to 0.10.2 (also cherry-picked commit e3bdc84d864a5ca655f985473770f7b7cb24b79b)
